### PR TITLE
Update style guide for new export and setget syntax

### PR DESCRIPTION
**Pull request by vmedea**
- `export` → `@export`
- `setget` → `:set(new_value)`

Also update to use new suggested syntax for emitting and connecting signals.

### CODE REVIEW

**Review by mhilbrunner — APPROVED** [review]

File: tutorials/scripting/gdscript/gdscript_styleguide.rst [hunk 1]
```diff
@@ -35,7 +35,7 @@ Here is a complete class example based on these guidelines:
 
     signal state_changed(previous, new)
 
-    export var initial_state = NodePath()
+    @export var initial_state := NodePath()
```
Comment (YuriSizov) [inline tutorials/scripting/gdscript/gdscript_styleguide.rst:38]: We should remove this example and export the node directly.

File: tutorials/scripting/gdscript/gdscript_styleguide.rst [hunk 2]
```diff
@@ -35,7 +35,7 @@ Here is a complete class example based on these guidelines:
...
     var is_active = true setget set_is_active
```
Comment (YuriSizov) [inline tutorials/scripting/gdscript/gdscript_styleguide.rst:39]: This is another `setget` example that we need to update. I think we can use a method reference here.

File: tutorials/scripting/gdscript/gdscript_styleguide.rst [hunk 3]
```diff
@@ -35,7 +35,7 @@ Here is a complete class example based on these guidelines:
...
 
     @onready var _state = get_node(initial_state) setget set_state
```
Comment (YuriSizov) [inline tutorials/scripting/gdscript/gdscript_styleguide.rst:41]: As mentioned above, should export a node directly here. And the setter needs to be updated to the new syntax.

**Review by YuriSizov — CHANGES_REQUESTED** [review]
This looks good to me, but there are more `setget` instances at the start of the article. We should also remove the NodePath example and export a node directly in there.

File: tutorials/scripting/gdscript/gdscript_styleguide.rst (hunk 1 above)
Comment (vmedea) [reply tutorials/scripting/gdscript/gdscript_styleguide.rst]: Updated to use a Node directly

**Comment by vmedea** [comment]
Updated according to the comments.
I've also added a commit that changes the use of signals to the currently suggested ways.

File: tutorials/scripting/gdscript/gdscript_styleguide.rst [hunk 4]
```diff
@@ -841,12 +845,12 @@ in that order.
...
         self._state = target_state
         _state.enter(msg)
-        Events.emit_signal("player_state_changed", _state.name)
+        Events.player_state_changed.emit(_state.name)
 
 
     func _on_state_changed(previous, new):
         print("state changed")
-        emit_signal("state_changed")
+        state_changed.emit()
```
Comment (vmedea) [inline tutorials/scripting/gdscript/gdscript_styleguide.rst:853]: I've only changed the syntax here, but functionality-wise this seems strange: the function that's connected to the signal re-emits the signal. I'd think the intent was to emit it in the setter instead.

**Comment by YuriSizov** [comment]
Sorry for the wait. I think some of these changes have been addressed by other PRs in the meantime. Would you mind rebasing this and checking if there is anything left to fix?

**Comment by vmedea** [comment]
Sure, updated.

**Review by mhilbrunner — APPROVED** [review]

**Comment by mhilbrunner** [comment]
Sorry we've been letting this one sit for so long! Merged and thanks for contributing :)

**Pull request merged**